### PR TITLE
Move flow validation to Nbworker topology.

### PR DESCRIPTION
**Pull request by dpoltavets**
Closes: #1442.

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SpeakerWorkerService.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Map;
+
+@Slf4j
+public class SpeakerWorkerService {
+    private final Map<String, SwitchId> keyToRequest = new HashMap<>();
+
+    /**
+     * Send command to speaker.
+     */
+    public void sendCommand(String key, SwitchId switchId, SpeakerWorkerCarrier carrier) {
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SpeakerWorkerService.java]: In the next PR's this worker will send dump meter command and update meter command. I see no reason to rename it now.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterBolt.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package org.openkilda.wfm.topology.nbworker.commands;
+
+import org.openkilda.messaging.command.CommandData;
+
+import lombok.Value;
+
+@Value
+public class RemoveKeyRouterBolt extends CommandData {
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterBolt.java]: Renamed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SpeakerWorkerService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/SpeakerWorkerService.java]: Renamed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java [hunk 3]
```diff
@@ -35,21 +36,48 @@
...
 public class RouterBolt extends AbstractBolt {
+    public static final String INCOME_STREAM = "router.bolt.stream";
+    private Set<String> keys = new HashSet<>();
 
     @Override
     protected void handleInput(Tuple input) throws PipelineException {
         String key = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
         Message message = pullValue(input, FIELD_ID_PAYLOAD, Message.class);
 
+        if (keys.contains(KeyProvider.getParentKey(key))) {
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java]: Why can't we route a request based on type/class?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/NbWorkerTopology.java [hunk 4]
```diff
@@ -73,6 +82,12 @@
     private static final String DISCO_KAFKA_BOLT_NAME = "disco-kafka-bolt";
     private static final String HISTORY_BOLT_NAME = "history-operations-bolt";
     private static final String NB_SPOUT_ID = "nb-spout";
+    private static final String SPEAKER_KAFKA_BOLT = "speaker-bolt";
+
+    private static final int HUB_TIMEOUT_MS = 13000;
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/NbWorkerTopology.java]: please add comment here or update pr as discussed

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java (hunk 3 above)
Comment (timofei-durakov) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java]: it looks like this is already resolved by field-grouping on the storm level, try to remove the keys at all

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterCommand.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+package org.openkilda.wfm.topology.nbworker.commands;
+
+import org.openkilda.messaging.command.CommandData;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveKeyRouterCommand extends CommandData {
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterCommand.java]: might be useless

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                builder.newStateMachine(FlowValidationState.INITIALIZED, carrier, key, request,
+                        persistenceManager, flowResourcesConfig);
+        process(fsm);
+    }
+
+    /**
+     * Handle response from speaker worker.
+     */
+    public void handleAsyncResponse(String key, Message message) {
+        FlowValidationFsm fsm = fsms.get(key);
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java]: check this is not null

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        }
+
+        process(fsm);
+    }
+
+    /**
+     * Handle timeout event.
+     */
+    public void handleTaskTimeout(String key) {
+        FlowValidationFsm fsm = fsms.get(key);
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java]: check this is not null

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java (hunk 3 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java]: Fixed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/NbWorkerTopology.java (hunk 4 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/NbWorkerTopology.java]: This has been moved to NbWorkerTopologyConfig.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterCommand.java (hunk 5 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/commands/RemoveKeyRouterCommand.java]: Removed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java (hunk 6 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java]: The check was added.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java (hunk 7 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowValidationHubService.java]: The check was added.

**Pull request merged**